### ISSUE TITLE
How to handle non-JSON responses based on Handler?

### ISSUE THREAD

**Issue by xeoncross** [mustafaakin/gongular#18]
For some of my http.Handlers I would like to have validation (or other errors) sent back inside a `html/template` (instead of simply output as JSON).

The Context struct is passed to the error handler.

    var defaultErrorHandler = func(err error, c *gongular.Context) {`

My first thought was to figure out how to use the `Context` as a middleman so that I could look for a flag set by the controller/handler. Then I could write an ErrorHandler that would return JSON or another format depending on if a flag was set. However, `c *gongular.Context` does not seem to provide such a flagging mechanism.

What is the correct way to support outputs other than JSON? Is there a way for a middleware to handle this?

*Update*: it looks like [a middleware can call `.StopChain()`](https://github.com/mustafaakin/gongular/blob/master/router.go#L168) to avoid the ErrorHandler. However, you can't seem to access other handlers/middlewares from a middleware.

```
type multiParam struct {
	Param struct {
		UserID string
		Page   int
	}
}

func (m *multiParam) Handle(c *gongular.Context) error {
	fmt.Println("multiparam")
	c.SetBody(fmt.Sprintf("%s:%d", m.Param.UserID, m.Param.Page))
	return nil
}

// The middleware to intercept validation errors
type middlewareHandler struct {}

func (m *middlewareHandler) Handle(c *gongular.Context) error {
	fmt.Println("middleware")
	return nil
}

func main() {
	e := newEngineTest()
	g := e.GetRouter().Group("/user/:UserID/page", &middlewareHandler{})
	g.GET("/:Page", &multiParam{})
}
```

**Comment by mustafaakin** [comment]
Exposing a regular context.Context through gongular.Context might be a good idea, what do you think?

**Comment by xeoncross** [comment]
Yes, that is one solution. `context.Context` would provide even more features than simply a shared request param map attached to `gongular.Context`.

Either would provide a way for downstream middleware to be notified of something.
